eox_tagging/edxapp_accessors.py

```diff
@@ -55,16 +58,33 @@ def get_course_enrollment(**kwargs):
     return CourseEnrollment.objects.get(user_id=user.id, course_id=course_id)
 
 
+def get_certificate(**kwargs):
+    """
+    Get GeneratedCertificate for specified id, key download URL or the course_id
+    and username associated with it.
+    """
+    username = kwargs.get("username")
+    course_id = kwargs.get("course_id")
+
+    return GeneratedCertificate.objects.get(
+        user__username=username,
+        course_id=CourseKey.from_string(course_id),
+    )
+
+
 def get_object(related_object_type, **kwargs):
```

Comment:
This functions is really hard to parse.

You can maybe rename it to `get_related_object` an rename the arg to `type` then use a chained `if..elif` block to call the correct getter.
```python
if type == 'user':
    get_user(**kwargs)
elif type == 'courseoverview':
    get_course(**kwargs)
elif type == 'site':
    get_site(**kwargs)
elif type == 'courseenrollment':
    get_course_enrollment(**kwargs)
elif type == 'generatedcertificate':
    get_certificate(**kwargs)
```